### PR TITLE
Temporary: Create a masterbar cart app

### PR DESCRIPTION
**Pull request by sirbrillig**
#### Changes proposed in this Pull Request

This is a branch of https://github.com/Automattic/wp-calypso/pull/52890 which creates a masterbar cart app for use in wp-admin.

masterbar-cart-app

#### Testing instructions

```
cd apps/masterbar-cart
yarn && yarn start
```

Visit http://localhost:3000/?site=[your_site_id_here]

### CODE REVIEW

File: client/layout/masterbar/masterbar-cart.tsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+	const onClick = () => {
+		setIsActive( ( active ) => {
+			if ( ! active ) {
+				reloadFromServer();
+			}
+			return ! active;
+		} );
+	};
+	const onClose = () => setIsActive( false );
+	const tooltip = translate( 'View my Shopping Cart' );
```
Comment (a8ci18n) [inline client/layout/masterbar/masterbar-cart.tsx:35]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **34** times:
`translate( 'My Shopping Cart' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/try/masterbar-cart-app)

**Comment by sirbrillig** [comment]
We should try to get the masterbar cart button styles applied in this app. Here's an example of a similar app that I based this PR on that may have the answer to that: https://github.com/Automattic/wp-calypso/pull/55189

Also we need to take a bunch of these commits that reduce the dependencies on WPLineItem and make them into their own PRs. I created https://github.com/Automattic/wp-calypso/pull/56383 to handle that.

**Comment by sirbrillig** [comment]
Next step: figure out how to actually inject this into the wp-admin masterbar.

**Comment by sirbrillig** [comment]
I removed the dependency on `WPOrderReviewLineItems` which should greatly reduce the size of the app since it won't need to pull in Redux.

**Comment by sirbrillig** [comment]
I'm going to close this and just use https://github.com/Automattic/wp-calypso/pull/52890 as the place to work on this for now.

**Pull request closed without merging**